include/core/Partition_Parser.hpp

```diff
@@ -0,0 +1,168 @@
+#ifndef PARTITION_PARSER_H
+#define PARTITION_PARSER_H
+
+#ifdef NGEN_MPI_ACTIVE
+
+//#include <mpi.h>
+
+#include <memory>
+#include <sstream>
+#include <tuple>
+#include <functional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <FeatureBuilder.hpp>
+#include "features/Features.hpp"
+#include <FeatureCollection.hpp>
+#include "JSONProperty.hpp"
+
+//This struct is moved from private section to here so that the unit test function can access it
+ struct PartitionData
+ {
+    int mpi_world_rank;
+    std::vector<std::string> cat_ids;
+    std::vector<std::string> nex_ids;
+    std::vector<std::string> remote_nexi;
```

Comment:
Would you like them to be together in a sub structure?

Reply:
Oh, I think we could make a std::vector<struct {remote_nexi, remote_catchments, remote_mpi_ranks}>. This might be what you had in mind.

Reply:
I'll add that for the sake of expediency, I am ok with leaving this to be fixed later (assuming it even needs to be), as long as we are confident the nexuses are being partitioned and parsed appropriately.  My concern is primarily whether it is possible that they are not, in some subtle way that is easy to miss.

Reply:
Copying some of @stcui007's comments from a different review item that pertain to this one:

> The array are not needed to be the same size.
> Thanks for the careful check on the variable nex-ids. Indeed it currently used anywhere in the code. I did not delete it from the code thinking just in case we might need local nexuses in the future. This can be changed.
> 
> Sorry, I meant nex-ids is currently not used anywhere in the code.

It sounds like this might need to be cleaned up in the future then, but it isn't causing any problems now.  I'll go ahead and mark this review item resolved.

Reply:
I'm taking @donaldwj's last comment on this issue to mean that these are actually needed by the _MPI_Network_ class.  Unresolving this review item.  Please clarify if this was incorrect.